Data/DataLoader.cs

```diff
@@ -14,6 +14,15 @@ public static List<Recommendation> LoadAll(string jsonFilePath)
             var json = File.ReadAllText(jsonFilePath);
             var opts = new JsonSerializerOptions { PropertyNameCaseInsensitive = true };
             var list = JsonSerializer.Deserialize<List<Recommendation>>(json, opts) ?? new List<Recommendation>();
+
+            // Precompute lowercase helper fields to speed up case-insensitive substring searches
+            foreach (var r in list)
+            {
+                r.NameLower = r.Name?.ToLowerInvariant() ?? string.Empty;
+                r.TheBestOfferLower = r.TheBestOffer?.ToLowerInvariant() ?? string.Empty;
+                r.NoteTipLower = r.NoteTip?.ToLowerInvariant() ?? string.Empty;
+            }
```

Comment:
This is a bit of code for something that already happens in the .net framework: changing case.  I'd leave it as is.
More code is clutter; less code is better.